x-pack/plugins/maps/common/content_management/v1/types.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  GetIn,
+  GetResult,
+  CreateIn,
+  CreateResult,
+  SearchIn,
+  SearchResult,
+  UpdateIn,
+  UpdateResult,
+  DeleteIn,
+  DeleteResult,
+} from '@kbn/content-management-plugin/common';
+import { MapContentType } from '../types';
+
+interface Reference {
+  type: string;
+  id: string;
+  name: string;
+}
+
+/* eslint-disable-next-line @typescript-eslint/consistent-type-definitions */
+export type MapSavedObjectAttributes = {
+  title: string;
+  description?: string;
+  mapStateJSON?: string;
+  layerListJSON?: string;
+  uiStateJSON?: string;
+};
+
+export interface MapItem {
+  id: string;
+  type: string;
+  version?: string;
+  createdAt?: string;
+  updatedAt?: string;
+  error?: {
+    error: string;
+    message: string;
+    statusCode: number;
+    metadata?: Record<string, unknown>;
+  };
+  attributes: MapSavedObjectAttributes;
```

Comment:
Good point. Done in https://github.com/elastic/kibana/pull/153304/commits/8f0949a876a35620eea3113ec200fc0c0e6d91a7

> but long term we probably should have these

Yes ideally in the future this will be refactored. 👍